### PR TITLE
fix: Consider folder state for vfolder invitations

### PR DESCRIPTION
**Pull request by ironAiken2**
### TL;DR

This PR is about considers the state of the vfolder for vfolder invitations.

> This PR must be merged after the corresponding PR on the core side is merged. https://github.com/lablup/backend.ai/pull/2780

### What changed?

- The basic logic of the webui is the same. However, I added logic to refresh the invitation on error. 
- Added playwrite test code to test vfolder invitations. **All tests are performed against the Backend.AI core and webui initial installations**.

### How to test?

1. Navigate to the Core PR branch and apply the changes.
2. Run Playwrite test code, or test it yourself. 

(if you're testing it yourself. It needs 2 browser)

1. Create a vfolder and invite other users to join.
2. Login as the invited user and verify that the invitation appears on the summary page.
3. Open the browser that sent the invite and move the vfolder to the trash.
4. Open the invited user's browser, accept the request, and verify that the invitation refreshes and displays the error.
---

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#2667** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2667?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: e2e/vfolder.test.ts [hunk 1]
```diff
@@ -1,11 +1,85 @@
...
 test.describe('VFolder ', () => {
   test('User can create and delete vFolder', async ({ page }) => {
     await loginAsUser(page);
     await createVFolderAndVerify(page, 'e2e-test-folder');
     await deleteVFolderAndVerify(page, 'e2e-test-folder');
   });
+
+  test('User can rename vfolder', async ({page}) => {
+    await loginAsUser(page);
+    await createVFolderAndVerify(page, 'e2e-rename-test-folder');await page.getByLabel('create', { exact: true }).click();
```
Comment (gahyuun) [inline e2e/vfolder.test.ts]: `await page.getByLabel('create', { exact: true }).click();`
This code occurs errors when I have multiple folders
`   Error: strict mode violation: getByLabel('create', { exact: true }) resolved to 2 elements:`

File: e2e/vfolder.test.ts [hunk 2]
```diff
@@ -1,11 +1,85 @@
...
+    await createVFolderAndVerify(page1, 'e2e-share-test-folder');
+    //share vfolder with user
+    await page1.getByRole('button', { name: 'share' }).click();
+    await page1.getByLabel('Enter E-Mail address').click();
+    await page1.getByLabel('Enter E-Mail address').fill("user2@lablup.com");
+    await page1.getByLabel('Edit+Delete').check();
+    await page1.locator('#share-button').getByLabel('share').click();
+    //accept shared vfodler
+    await loginAsUser2(page2);
+    await page2.getByLabel('Accept').click();
```
Comment (gahyuun) [inline e2e/vfolder.test.ts]: `await page2.getByLabel('Accept').click();`
This code occurs errors when I have multiple folders
   Error: strict mode violation: getByLabel('Accept') resolved to 2 elements:

**Review by gahyuun — CHANGES_REQUESTED** [review]

File: e2e/vfolder.test.ts [hunk 3]
```diff
@@ -1,11 +1,94 @@
...
+    //share vfolder with user
+    await page1.getByRole('button', { name: 'share' }).click();
+    await page1.getByLabel('Enter E-Mail address').click();
+    await page1.getByLabel('Enter E-Mail address').fill("user2@lablup.com");
+    await page1.getByLabel('Edit+Delete').check();
+    await page1.locator('#share-button').getByLabel('share').click();
+    //check vfolder invitation was sent
+    await loginAsUser2(page2);
+    await page2.waitForTimeout(1000);
+    expect(page1.getByTitle('Invitation').locator('#activity').nth(0));
```
Comment (gahyuun) [inline e2e/vfolder.test.ts]: That code finds the element but doesn't validate any conditions like `toBeVisible`.
Also when there are multiple invitations, `nth(0)` seems to be a different folder invitation than `e2e-share-test-folder.  `

**Review by gahyuun — CHANGES_REQUESTED** [review]
Adding the `page.waitForTimeout(3000);` to `deleteVFolderAndVerify` seems to always succeed. 
  
```
...
  await page.locator('#delete-without-confirm-button').getByLabel('delete').click();
  await page.waitForTimeout(3000);
...
```

**Review by lizable — CHANGES_REQUESTED** [review]
LGTM.

**Review by lizable — APPROVED** [review]
LGTM.
